File: py/day1.py
import functools

def split_items(string):
  return string.split("\n")

f = open("input.txt", "r")
lines = f.readlines()
items_elf = list(
  map(split_items,
      functools.reduce(lambda x, y: x + y, lines).split("\n\n")))
elves = []
for i in items_elf:
  ints = list(map(int, i))
  elves.append(functools.reduce(lambda x, y: x + y, ints))
elves.sort(reverse=True)
silver = elves[0]
gold = elves[0] + elves[1] + elves[2]
print("silver:", silver)
print("gold:", gold)

File: rs/day1.rs
use std::fs;

fn main() {
  let contents = fs::read_to_string("input.txt").expect("Should have been able to read the file");
  let split = contents.split("\n\n");
  let mut elves = split
    .map(|s|
      s.split("\n")
        .map(|s| s.parse::<i32>().unwrap())
        .sum::<i32>()
    )
    .collect::<Vec<i32>>();
  elves.sort();
  elves.reverse();
  let silver = elves[0];
  let gold = elves[0] + elves[1] + elves[2];
  println!("silver: {:?}", silver);
  println!("gold: {:?}", gold);
}